plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/nodespacing/NodeLabelAndSizeCalculator.java

```diff
@@ -0,0 +1,247 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Kiel University and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Kiel University - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.elk.core.util.nodespacing;
+
+import org.eclipse.elk.core.math.ElkPadding;
+import org.eclipse.elk.core.util.adapters.GraphAdapters.GraphAdapter;
+import org.eclipse.elk.core.util.adapters.GraphAdapters.NodeAdapter;
+import org.eclipse.elk.core.util.nodespacing.internal.NodeContext;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.CellSystemConfigurator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.HorizontalPortPlacementSizeCalculator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.InsidePortLabelCellCreator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.LabelPlacer;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.NodeLabelAndSizeUtilities;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.NodeLabelCellCreator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.NodeSizeCalculator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.PortContextCreator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.PortLabelPlacementCalculator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.PortPlacementCalculator;
+import org.eclipse.elk.core.util.nodespacing.internal.algorithm.VerticalPortPlacementSizeCalculator;
+import org.eclipse.elk.core.util.nodespacing.internal.cellsystem.Cell;
+import org.eclipse.elk.core.util.nodespacing.internal.cellsystem.ContainerArea;
+import org.eclipse.elk.core.util.nodespacing.internal.cellsystem.GridContainerCell;
+
+/**
+ * Knows how to calculate the size of a node and how to place its ports. Takes all size constraints and options into
+ * account.
+ * 
+ * <p>This calculator internally understands the layout of the node as a grid-like system of cells which different
```

Comment:
Nope, but I changed it to "...in which different labels and ports are placed.", which should be a bit clearer.